ocaml/typing/ctype.ml

```diff
@@ -4777,8 +4811,10 @@ let rec moregen inst_nongen variance type_pairs env t1 t2 =
                 || !Clflags.classic && equivalent_with_nolabels l1 l2) ->
               moregen inst_nongen (neg_variance variance) type_pairs env t1 t2;
               moregen inst_nongen variance type_pairs env u1 u2;
-              moregen_alloc_mode (neg_variance variance) a1 a2;
-              moregen_alloc_mode variance r1 r2
+              (* [t2] and [u2] is the user-written interface, which we deem as
```

Comment:
I don't understand this comment -- what is the impact of this choice? It might be good to e.g. have a test that demonstrates the difference in behavior between choosing the `2` types vs the `1` types here. Then we could refer to the test from this comment to illustrate the point.